slick/src/main/scala/slick/basic/BasicBackend.scala

```diff
@@ -368,6 +374,14 @@ trait BasicBackend { self =>
 
     private[BasicBackend] var currentSession: Session = null
 
+    private[BasicBackend] var releasedConnection = false
+
+    private[BasicBackend] def priority(highPriority: Boolean): Priority = {
```

Comment:
This looks a bit confusing. Can we find better names for the priority levels? Maybe something like HasConnection / Continuation / New?